cinema_project/base_app/models.py

```diff
@@ -54,22 +60,57 @@ def __str__(self):
 class Schedule(models.Model):
     movie = models.ForeignKey(
         Movie, default=None, on_delete=models.SET_DEFAULT)
+    schedule_time = models.DateTimeField(default=datetime(2022, 1, 1, 1, 1, 1, 1))
+
+    hall = models.ForeignKey(
+        Hall, on_delete=models.CASCADE, null=True
+    )
+
+    def __str__(self):
+        return f"{self.movie} {self.hall} {self.schedule_time} {self.hall.cinema.city}"
+
+
+class Seat(models.Model):
     hall = models.ForeignKey(
-        Hall, default=None, on_delete=models.SET_DEFAULT)
-    schedule_time = models.DateTimeField(default=datetime.now())
+        Hall, on_delete=models.CASCADE, null=True
+    )
+    name = models.CharField(max_length=100, default=None)
 
     def __str__(self):
-        return f"{self.hall}__{self.schedule_time}"
+        return self.name
 
 
 class Reservation(models.Model):
     user = models.ForeignKey(
         User, default=None, on_delete=models.CASCADE
     )
+    seat = models.ForeignKey(Seat, on_delete=models.CASCADE, default=None)
     schedule = models.ForeignKey(
         Schedule, default=None, on_delete=models.CASCADE
     )
+    is_confirmed = models.BooleanField(default=False)
 
+    def __str__(self):
+        return f"{self.user}__{self.seat}__{self.schedule.schedule_time}"
 
-class Seat(models.Model):
-    reservation = models.ForeignKey(Reservation, default=None, on_delete=models.CASCADE)
+    class Meta:
+        unique_together = ['seat', 'schedule']
+
+
+def send_email_when_movie_is_created(sender, instance, *args, **kwargs):
+    all_users = [user.email for user in User.objects.all()]
+    movie_name = instance.name
+    movie_duration = instance.duration
+    movie_director = instance.director
+
+    for user_email in all_users:
+        send_email(from_email="test@test.ro",
+                   subject=f'{movie_name} added to Cinema X',
+                   to_email=f"{user_email}",
+                   html_content=f"<p>{movie_name} now at Cinema X</p>"
+                                f"<p>Directed by: {movie_director}</p>"
+                                f"<p>Duration: {movie_duration}</p>")
+    pass
+
+
+post_save.connect(send_email_when_movie_is_created, sender=Movie)
```

Comment:
This will send emails each time `.save()` is called, which means editing a movie will also send an email.